From ec7756155b3661a9a8b61a1deeba03cea02c9dd9 Mon Sep 17 00:00:00 2001
From: Daniel Golle <daniel@makrotopia.org>
Date: Thu, 16 Mar 2023 20:00:33 +0530
Subject: [PATCH 157/500] OpenWrt:
 725-net-phy-aquantia-add-PHY_IDs-for-AQR112-variants.patch

net: phy: aquantia: add PHY_ID for AQR112R

As advised by Ian Chang this PHY is used in Puzzle devices.

Signed-off-by: Daniel Golle <daniel@makrotopia.org>

Change-Id: Ice3eaeff569aedc9200997f80d0dfdcb7a679ddf
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit 031135e325fee9b59ab94c6d9cf2273c5afea34d)
---
 drivers/net/phy/aquantia_main.c | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/drivers/net/phy/aquantia_main.c b/drivers/net/phy/aquantia_main.c
index c9061676155e..1c5348ff5027 100644
--- a/drivers/net/phy/aquantia_main.c
+++ b/drivers/net/phy/aquantia_main.c
@@ -29,6 +29,8 @@
 #define PHY_ID_AQR112	0x03a1b662
 #define PHY_ID_AQR412	0x03a1b712
 #define PHY_ID_AQR113	0x31c31c40
+#define PHY_ID_AQR112C  0x03a1b790
+#define PHY_ID_AQR112R  0x31c31d12
 
 #define MDIO_PHYXS_VEND_IF_STATUS		0xe812
 #define MDIO_PHYXS_VEND_IF_STATUS_TYPE_MASK	GENMASK(7, 3)
@@ -1028,6 +1030,31 @@ static struct phy_driver aqr_driver[] = {
 	.handle_interrupt = aqr_handle_interrupt,
 	.read_status	= aqr107_read_status,
 },
+{
+       PHY_ID_MATCH_MODEL(PHY_ID_AQR112C),
+       .name           = "Aquantia AQR112C",
+       .probe          = aqr107_probe,
+       .config_aneg    = aqr_config_aneg_set_prot,
+       .config_intr    = aqr_config_intr,
+       .handle_interrupt = aqr_handle_interrupt,
+       .read_status    = aqr107_read_status,
+       .get_sset_count = aqr107_get_sset_count,
+       .get_strings    = aqr107_get_strings,
+       .get_stats      = aqr107_get_stats,
+},
+{
+       PHY_ID_MATCH_MODEL(PHY_ID_AQR112R),
+       .name           = "Aquantia AQR112R",
+       .probe          = aqr107_probe,
+       .config_aneg    = aqr_config_aneg_set_prot,
+       .config_intr    = aqr_config_intr,
+       .handle_interrupt = aqr_handle_interrupt,
+       .read_status    = aqr107_read_status,
+       .get_sset_count = aqr107_get_sset_count,
+       .get_strings    = aqr107_get_strings,
+       .get_stats      = aqr107_get_stats,
+},
+
 };
 
 module_phy_driver(aqr_driver);
@@ -1045,6 +1072,8 @@ static struct mdio_device_id __maybe_unused aqr_tbl[] = {
 	{ PHY_ID_MATCH_MODEL(PHY_ID_AQR113C) },
 	{ PHY_ID_MATCH_MODEL(PHY_ID_AQR813) },
 	{ PHY_ID_MATCH_MODEL(PHY_ID_AQR113) },
+	{ PHY_ID_MATCH_MODEL(PHY_ID_AQR112C) },
+	{ PHY_ID_MATCH_MODEL(PHY_ID_AQR112R) },
 	{ }
 };
 
-- 
2.34.1

